src/v/iceberg/datatypes_json.cc

```diff
@@ -0,0 +1,251 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/datatypes_json.h"
+
+#include "iceberg/datatypes.h"
+#include "iceberg/json_utils.h"
+#include "json/document.h"
+#include "strings/string_switch.h"
+
+#include <stdexcept>
+#include <string>
+
+namespace iceberg {
+
+namespace {
+std::string_view
+extract_between(char start_ch, char end_ch, const std::string_view& s) {
```

Comment:
>const std::string_view& s

std::string_view is already a `const` reference to a string. passing by reference may even be slower https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/#1-eliminate-a-pointer-indirection-in-the-callee

Reply:
Done